internal/products.go

```diff
@@ -48,6 +48,39 @@ type Product struct {
 	Description  string       `json:"description"`
 }
 
+// Take the "Product" as returned from an RMT and adjust the repository URLs
+// to include a "credentials" parameter. This is somewhat specific to the RMT
+// instances available for Public Cloud on-demand instances, as they need to
+// authenticate to be able to access repositories.
+func fixRepoUrlsForRMT(p *Product) error {
+	for i := range p.Repositories {
+		repourl, err := url.Parse(p.Repositories[i].URL)
+		if err != nil {
+			loggedError("Unable to parse repository URL: %s - %v", p.Repositories[i].URL, err)
+			return err
+		}
+		params := repourl.Query()
+		if err != nil {
```

Comment:
There is no new error assignment above than the one in line 57.

Reply:
fixed